### ISSUE TITLE
Duplicated definition of built-in magic aliases after %reset

### ISSUE THREAD

**Issue by red8012** [ipython/ipython#11646]
When running `%reset -f` in Jupyter notebook or Spyder, the following message will be printed.
```
ERROR:root:Invalid alias: The name clear can't be aliased because it is another magic command.
ERROR:root:Invalid alias: The name more can't be aliased because it is another magic command.
ERROR:root:Invalid alias: The name less can't be aliased because it is another magic command.
ERROR:root:Invalid alias: The name man can't be aliased because it is another magic command.
```
However, running `%reset -f` in IPython itself is just fine.

I started experiencing this after upgrading to IPython 7.3.0, and I believe that this behavior was from the change #11528. Maybe those aliases should only be redefined when running in the IPython shell, not in Jupyter or Spyder as a kernel.

**Comment by ccordoba12** [comment]
@Carreau, any hope to fix this problem? It's very annoying for Spyder users.

**Comment by Carreau** [comment]
Likely. Worst case I revert  #11528. Though I'm stretched thin trying to maintain IPython on my free time. It's not my fulltime job anymore, so i can't commit as much time as before.

**Comment by ccordoba12** [comment]
Ok, I understand. Thanks for the quick response.

**Comment by texadactyl** [comment]
Successful work-around in Anaconda environment:
`conda install ipython=7.2.0`

**Comment by texadactyl** [comment]
Still an issue in ipython-7.4.0

**Comment by hrishikeshrt** [comment]
7.5 the issue remains

**Comment by astrofrog** [comment]
I'm running into this too

**Comment by daniel-lorenzo** [comment]
Try this:    %reset -s -f

**Comment by texadactyl** [comment]
In Spyder, using ipython 7.5, every execution still shows this at the beginning:
ERROR:root:Invalid alias: The name clear can't be aliased because it is another magic command.
ERROR:root:Invalid alias: The name more can't be aliased because it is another magic command.
ERROR:root:Invalid alias: The name less can't be aliased because it is another magic command.
ERROR:root:Invalid alias: The name man can't be aliased because it is another magic command.

My new conda update procedure:
conda update --all
conda install ipython=7.2.0

I'd like to stay up-to-date in ipython.  Continued use of version 7.2.0 seems like the "lesser of evils".

**Comment by mx73** [comment]
Still an issue in ipython 7.6

**Comment by Carreau** [comment]
> Still an issue in ipython 7.6

Yes, the issue has not been marked as closed.

**Comment by PawanKuma** [comment]
> In Spyder, using ipython 7.5, every execution still shows this at the beginning:
> ERROR:root:Invalid alias: The name clear can't be aliased because it is another magic command.
> ERROR:root:Invalid alias: The name more can't be aliased because it is another magic command.
> ERROR:root:Invalid alias: The name less can't be aliased because it is another magic command.
> ERROR:root:Invalid alias: The name man can't be aliased because it is another magic command.
> 
> My new conda update procedure:
> conda update --all
> conda install ipython=7.2.0
> 
> I'd like to stay up-to-date in ipython. However, continued use of version 7.2.0 seems like the "lesser of evils".

This worked for me thanks!!

**Comment by texadactyl** [comment]
@PawanKuma You are welcome.

**Comment by texadactyl** [comment]
@Carreau Thank you very much.

**Comment by Carreau** [comment]
@juanis2112 did the hard part of the work.

**Comment by texadactyl** [comment]
@juanis2112 
Good job. Much thanks.

**Comment by irvanseptiar** [comment]
I also got this issue in ipython 7.6

**Comment by texadactyl** [comment]
@irvanseptiar 
Please read the entire thread and you will see that the issue was resolved in version 7.7.
Update your repo.

**Comment by ccordoba12** [comment]
Thanks a lot @juanis2112!
